[fix](ES Catalog)Only push down literal expr in binary predicate(#39845)

qidaye opened the pull request:
## Proposed changes

This only affect old planner.  close #39849
bp #39845

qidaye commented:
run buildall

morrySnow commented on fe/fe-core/src/main/java/org/apache/doris/external/elasticsearch/QueryBuilders.java:
```diff
@@ -266,7 +267,21 @@ public static QueryBuilder toEsDsl(Expr expr, List<Expr> notPushDownList, Map<St
+            // only push down literal expr to ES
+            if (value instanceof LiteralExpr) {
+                LiteralExpr literalExpr = (LiteralExpr) value;
+                return parseBinaryPredicate(literalExpr, opCode, column, needDateCompat);
```
flip opcode if value from left side

qidaye committed:
[improve](ES Catalog)Only push down literal expr in binary predicate (#39845)

## Proposed changes

1. Add support for reverse binary predicate, such as ` 3 >= k3`.
2. Only push down literal expr in binary predicate
This only affects old planner, new planner dose not have any bug. 
We add check here to make the push down to ES more robust.

qidaye commented:
run buildall

airborne12 reviewed (approved):
LGTM
